yb-voyager/src/tgtdb/conn_pool.go

```diff
@@ -91,6 +93,26 @@ func (pool *ConnectionPool) WithConn(fn func(*pgx.Conn) (bool, error)) error {
 	return err
 }
 
+func (pool *ConnectionPool) CachePreparedStmtForConn(connId uint32, ps string) {
```

Comment:
I would suggest to completely hide the "cache" from the outside code. You can define a `ConnectionPool.PrepareStatement(conn, stmtName, stmt)` method. If the stmtName is already in the cache, the method will immediately return. Else, it will proceed to actual statement preparation. Whenever a connection is dropped, you can simply drop corresponding entry from the cache.